lxd/storage/backend_lxd.go

```diff
@@ -3147,6 +3155,15 @@ func (b *lxdBackend) EnsureImage(fingerprint string, op *operations.Operation) e
 	revert := revert.New()
 	defer revert.Fail()
 
+	imgVol.SetHasSource(false)
+
+	// The imgVol needs to have a config otherwise filesystem zvols will not be detected correctly,
+	// and will lack the `_<filesystem>` suffix.
+	err = b.driver.FillVolumeConfig(imgVol)
```

Comment:
So thinking more about this, I think I would prefer to only have `VolumeDBCreate` call `FillVolumeConfig` so it keeps it all together in one place.

For this to work we'd need to create the volume record before the storage volume itself, which is no bad thing as that is what is done elsewhere (and we can easily add a revert to delete the record of the volume creation fails).

Then whats left is that we would need to update the volume record once the the unpack has finished if the `volFiller.Size` is set, and for this we can use `b.state.DB.Cluster.UpdateStoragePoolVolume()`.

In this way you also won't need the call to `imgVol.SetHasSource(false)`.